draft-aegis-mimi-arch-latest.md

```diff
@@ -83,16 +221,19 @@ Bob invites a multi-device account of Alice. Searching by a tag 2.
 above allows to add an arbitrary moderator, or all clients owned by
 entities with the right clearance.
 
-## Registering Clients
+## Identity Registration
 When a client registers with an application, the application registers
-the client with the identity service using the following interface.
+the client with the identity service using the following query:
+
+// TODO: This should be described as an HTTP route.
 
 ~~~
-create_identity(
-    application_id: UUID,
-    client_identity: Blob, // The client's credential, e.g. and X509 certificate
-    tags: Tags // A list of tags used to search for identities of another entity
-) -> client_id: UUID 
+u16 IdentityQueryType = 1;
+
+struct {
+    opaque entity_identifier<V>;
+    opaque tag<V>;
+} RegisterIdentityQuery;

```

Comment:
Should there also be RegisterIdentityResponse?

Reply:
Yes, also, this should be HTTP + JSON. We need to figure out how we want to generally display request response. I asked around, there is no standard way of doing it.